### DISCUSSION TITLE
Google provider, expires_at always 16 hours in the past

### DISCUSSION THREAD

**Discussion by dprothero** [nextauthjs/next-auth#6168] [Help] [answered]
I'm using the Google provider and am trying to implement refresh tokens. However, when I inspect `account.expires_at` in the `jwt` callback like so...

```ts
  callbacks: {
    async jwt({ token, user, account, profile }) {
      // Initial sign in
      if (account && user) {
        console.debug('account: ', JSON.stringify(account, null, 2));
        console.debug('user: ', JSON.stringify(user, null, 2));
        console.debug('token: ', JSON.stringify(token, null, 2));
        console.debug('profile: ', JSON.stringify(profile, null, 2));
```

The `expires_at` is always 16 hours AGO. However, `profile.exp` holds the expected value of 1 hour in the future.

```json
account:  {
  "provider": "google",
  "type": "oauth",
  "providerAccountId": "REDACTED",
  "access_token": "REDACTED",
  "expires_at": 1671787049,    <--------
  "refresh_token": "REDACTED",
  "scope": "https://www.googleapis.com/auth/userinfo.email openid https://www.googleapis.com/auth/drive https://www.googleapis.com/auth/userinfo.profile",
  "token_type": "Bearer",
  "id_token": "REDACTED"
}
user:  {
  "id": "REDACTED",
  "name": "REDACTED",
  "email": "REDACTED",
  "image": "https://lh3.googleusercontent.com/a/REDACTED"
}
token:  {
  "name": "REDACTED",
  "email": "REDACTED",
  "picture": "https://lh3.googleusercontent.com/a/REDACTED",
  "sub": "REDACTED"
}
profile:  {
  "iss": "https://accounts.google.com",
  "azp": "REDACTED.apps.googleusercontent.com",
  "aud": "REDACTED.apps.googleusercontent.com",
  "sub": "REDACTED",
  "email": "REDACTED",
  "email_verified": true,
  "at_hash": "REDACTED",
  "name": "REDACTED",
  "picture": "https://lh3.googleusercontent.com/a/REDACTED",
  "given_name": "REDACTED",
  "family_name": "REDACTED",
  "locale": "en",
  "iat": 1671844432,   <------
  "exp": 1671848032  <------
}
```

So, looking at these three values:

```
expires_at: 1671787049
iat: 1671844432
exp: 1671848032
```

We see the token was issued at (`iat`) 1671844432, or "Saturday, December 24, 2022 1:13:52 AM" and according to the JWT's `exp` field, expires at 1671848032, or "Saturday, December 24, 2022 2:13:52 AM" (exactly one hour later).

So, why is `expires_at` 1671787049, which decodes to Friday, December 23, 2022 9:17:29 AM?

**Comment by dprothero** [comment]
I tried with the Dropbox provider and I get a value that's 13 hours in the past. There's got to be a bug somewhere in next-auth or an upstream library, yeah?

**Reply by vwatel** [reply]
Maybe your issue is related to some wrong config in the refresh token example - see https://github.com/nextauthjs/next-auth/pull/4849

**Reply by dprothero** [reply]
That's a very good point @vwatel. That certainly led to a fair amount of confusion, as well.

**Comment by dprothero** [comment, marked as answer]
I think this was related to the clock in the WSL virtual machine falling out of sync when the laptop sleeps. After fixing that, it seems to be working correctly.
